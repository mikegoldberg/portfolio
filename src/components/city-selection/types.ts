export interface CitySelectionProps {
  city: string;
  onSubmit: (name: string) => void;
}
