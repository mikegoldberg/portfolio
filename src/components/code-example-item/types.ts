import { MouseEventHandler } from "react";

export interface CodeExampleItemProps {
  title: string;
  poster: string;
  onClick: MouseEventHandler;
  source: string;
}
