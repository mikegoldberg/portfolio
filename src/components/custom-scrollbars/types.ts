import { ReactElement } from "react";

export interface CustomScrollbarsProps {
  onScrollFrame: (position: number) => void;
  children?: ReactElement;
}
