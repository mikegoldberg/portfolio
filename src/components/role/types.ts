type Title = string;
type Company = string;
type Description = string;
type Skillset = Array<string>;
type Achievements = Array<string>;
type Start = string;
type End = string | null;
type Contract = boolean;

export interface RoleProps {
  title: Title;
  company: Company;
  description: Description;
  skillset: Skillset;
  achievements: Achievements;
  start: Start;
  end: End;
  contract: Contract;
}
