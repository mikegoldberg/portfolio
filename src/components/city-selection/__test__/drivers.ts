export const HEADING = "city-selection-heading";
