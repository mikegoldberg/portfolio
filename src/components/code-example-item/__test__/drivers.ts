export const POSTER = "code-example-item-poster";
export const ONCLICK = "code-example-item-onclick";
export const TITLE = "code-example-item-title";
export const SOURCE = "code-example-item-source";
