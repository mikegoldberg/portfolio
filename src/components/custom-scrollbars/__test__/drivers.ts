export const CONTENT = "custom-scrollbars-content";
