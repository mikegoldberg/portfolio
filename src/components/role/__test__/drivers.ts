export const TITLE = "role-title";
export const COMPANY = "role-company";
export const DESCRIPTION = "role-description";
export const ACHIEVEMENTS = "role-achievements";
export const SKILLSET = "role-skillset";
export const START = "role-start";
export const END = "role-end";
