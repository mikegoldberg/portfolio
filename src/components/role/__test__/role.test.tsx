import { render } from "../../../test-utils";
import { screen } from "@testing-library/react";
import Role from "..";
import { RoleProps } from "../types";
import * as TEST_DRIVERS from "./drivers";

const defaultProps: RoleProps = {
  company: "company name",
  contract: true,
  title: "job title",
  start: "2022-03-01T12:00-0500",
  end: "2022-09-30T12:00-0500",
  description: "test description.",
  achievements: ["achievements 1", "achievements 2"],
  skillset: ["test skill 1", "test skill 2"],
};

describe("Role", () => {
  it("renders correctly", () => {
    const { container } = render(<Role {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("renders a title", () => {
    render(<Role {...defaultProps} />);
    const element = screen.queryByTestId(TEST_DRIVERS.TITLE);
    expect(element).not.toBeNull();
  });

  it("renders a description", () => {
    render(<Role {...defaultProps} />);
    const element = screen.queryByTestId(TEST_DRIVERS.DESCRIPTION);
    expect(element).not.toBeNull();
  });

  it("renders a achievements", () => {
    render(<Role {...defaultProps} />);
    const element = screen.queryByTestId(TEST_DRIVERS.ACHIEVEMENTS);
    expect(element).not.toBeNull();
  });

  it("renders the provided skillsets=", () => {
    render(<Role {...defaultProps} />);
    const element = screen.getByTestId(TEST_DRIVERS.SKILLSET);
    expect(element.children.length).toBe(defaultProps.skillset.length);
  });

  it("renders a start date", () => {
    render(<Role {...defaultProps} />);
    const element = screen.queryByTestId(TEST_DRIVERS.START);
    expect(element).not.toBeNull();
  });

  it("renders a end date", () => {
    render(<Role {...defaultProps} />);
    const element = screen.queryByTestId(TEST_DRIVERS.END);
    expect(element).not.toBeNull();
  });
});
