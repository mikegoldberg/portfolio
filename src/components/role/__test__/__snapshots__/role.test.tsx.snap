// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Role renders correctly 1`] = `
<div>
  <div
    class="css-1is0qkf"
  >
    <div
      class="css-jqwek2"
    >
      <p
        class="chakra-text css-5yjrf7"
        data-testid="role-title"
      >
        job title
      </p>
    </div>
    <div
      class="css-1w6vqyc"
    >
      <p
        class="chakra-text css-tjwf6p"
        data-testid="role-company"
      >
        company name
      </p>
      <div
        class="css-1ncd50l"
      >
        <time
          data-testid="role-start"
          datetime="1646154000000"
        >
          Mar 2022
        </time>
        <p
          class="chakra-text css-0"
        >
          -
        </p>
        <time
          data-testid="role-end"
          datetime="1664557200000"
        >
          Sep 2022
        </time>
        <p
          class="chakra-text css-0"
        >
          ( 6 mos)
        </p>
      </div>
    </div>
    <p
      class="chakra-text css-ezl6qi"
      data-testid="role-description"
    >
      test description.
    </p>
    <ul
      class="css-19kmp78"
      data-testid="role-achievements"
      role="list"
    >
      <li
        class="css-1msjh1x"
      >
        achievements 1
      </li>
      <li
        class="css-1msjh1x"
      >
        achievements 2
      </li>
    </ul>
    <div
      class="css-sq4ju6"
      data-testid="role-skillset"
    >
      <span
        class="css-1soqvl"
      >
        <span
          class="css-1adgxkr"
        >
          test skill 1
        </span>
      </span>
      <span
        class="css-1soqvl"
      >
        <span
          class="css-1adgxkr"
        >
          test skill 2
        </span>
      </span>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
